src/Moq/Interception/InterceptionAspects.cs

```diff
@@ -380,7 +380,7 @@ public static void Handle(Invocation invocation, Mock mock)
 
 							if (innerMock != null)
 							{
-								Mock.SetupAllProperties(innerMock);
+								Mock.SetupAllProperties(innerMock, mock.AutoSetupPropertiesDefaultValueProvider);
```

Comment:
I did it this way because I thought that the following lines
https://github.com/moq/moq4/blob/9ea157cf61f1eb7deeadc419cac08d1aa9fd4f44/src/Moq/Mock.cs#L553-L556

does the same in current implementation.

`defaultValueProvider` passed from
https://github.com/moq/moq4/blob/9ea157cf61f1eb7deeadc419cac08d1aa9fd4f44/src/Moq/Mock.cs#L497
is captured for inner mocks here.

That was my understanding. What did I miss here?

Reply:
>What did I miss here?

Probably nothing. It seems it is I who missed something. 😄

It is possible that the present-day implementation is faulty in that regard. I still think the root mock's provider shouldn't override inner mocks' ones. But that is probably a separate issue.

What counts more for the moment is that we're not introducing a functional change here. If your current implementation works the same way as present-day Moq (perhaps ensure that with a new test), then we can resolve this discussion.

Reply:
I just ran a quick test, your implementation acts the same way as present-day Moq. My bad, let's leave your implementation as it is now.

Reply:
I've added `Property_stubbed_by_SetupAllProperties_should_use_parent_DefaultValue_behaviour_for_inner_mocks` to capture current behavior in tests.